### ISSUE TITLE
Can't connect to the UI

### ISSUE THREAD

**Issue by shinji62** [goharbor/harbor#6889]
When I try to login on the UI, the login button get gray and nothing happen.
Sometimes it work, especially when I restart the core pod.

I try with version 1.7.0 but issue is the same..
I am using the HA settings including Redis and Postgres, I can see data in Redis and Postgres so I assume they are working.
I am behind Nginx Ingress.

**Steps to reproduce the problem:**
Please provide the steps to reproduce this problem.

Go to https://domain.com/harbor/sign-in then try to login. Nothing happen just the button goes gray.

In the core logs I see 
```
2019-02-04T07:16:41Z [DEBUG] [security.go:280]: can not get user information from session
2019-02-04T07:16:41Z [DEBUG] [security.go:337]: user information is nil
2019-02-04T07:16:41Z [DEBUG] [security.go:351]: using local database project manager
2019-02-04T07:16:41Z [DEBUG] [security.go:353]: creating local database security context...
2019-02-04T07:16:43Z [DEBUG] [security.go:280]: can not get user information from session
2019-02-04T07:16:43Z [DEBUG] [security.go:337]: user information is nil
2019-02-04T07:16:43Z [DEBUG] [security.go:351]: using local database project manager
2019-02-04T07:16:43Z [DEBUG] [security.go:353]: creating local database security contex
```

In the UI logs
```
211.122.24.78 - [211.122.24.78] - - [04/Feb/2019:07:16:40 +0000] "POST /c/login HTTP/1.1" 200 0 "https://domain/harbor/sign-in?signout=true" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/72.0.3626.81 Safari/537.36" 868 0.014 [harbor-harbor-harbor-core-80] IP:8080 0 0.013 200 b8b11dced1077592ec92a95415ce77eb

```
Using Docker client is always working.

Seems postgres is connected and Redis too.
```
019-02-04T06:56:27Z [DEBUG] [init.go:37]: topic OnDeletion is subscribed
2019-02-04T06:56:27Z [DEBUG] [init.go:37]: topic StartReplication is subscribed
2019-02-04T06:56:27Z [DEBUG] [init.go:37]: topic OnPush is subscribed
2019-02-04T06:56:27Z [INFO] Config path: /etc/core/app.conf
2019-02-04T06:56:27Z [INFO] initializing configurations...
2019-02-04T06:56:27Z [INFO] key path: /etc/core/key
2019-02-04T06:56:27Z [INFO] initializing client for adminserver http://harbor-harbor-adminserver ...
2019-02-04T06:56:27Z [INFO] initializing the project manager based on local database...
2019-02-04T06:56:27Z [INFO] configurations initialization completed
2019-02-04T06:56:27Z [INFO] Registering database: type-PostgreSQL host-IP port-5432 databse-tooling-prod-registry-db sslmode-"require"
2019-02-04T06:56:27Z [INFO] Register database completed
2019-02-04T06:56:27Z [INFO] User id: 1 already has its encrypted password.
2019-02-04T06:56:27Z [INFO] Enable redis cache for chart caching
2019-02-04T06:56:27Z [DEBUG] [chart_repository.go:503]: Chart storage server is set to http://harbor-harbor-chartmuseum
2019-02-04T06:56:27Z [INFO] API controller for chart repository server is successfully initialized
2019-02-04T06:56:27Z [INFO] initialized clair database
2019-02-04T06:56:27Z [INFO] Because SYNC_REGISTRY set false , no need to sync registry
2019-02-04T06:56:27Z [INFO] Init proxy
2019/02/04 06:56:27 [I] [asm_amd64.s:2337] http server Running on http://:8080
```

**Versions:**
Please specify the versions of following systems.
- harbor version: [dev]
- Kubernetes GKE

**Comment by Morlackx** [comment]
I have exactly the same issue here (behaviors, logs, everything).

Seems that login works (if I try to logon with a wrong one I get Invalid credentials), but any other calls are unauthorized :

With wrong user/pass:
![screenshot from 2019-02-05 12-38-52](https://user-images.githubusercontent.com/4631115/52271115-18f88a80-2943-11e9-9819-a34cef6900be.png)

WIth good credentials:
![screenshot from 2019-02-05 12-37-38](https://user-images.githubusercontent.com/4631115/52271143-30d00e80-2943-11e9-8bfb-6c381eaad165.png)

It is deployed on a k8s on premises.
Harbor 1.7.0 (Helm chart 1.0.0) 
DB: Stolon 0.13-pg10 (Helm chart stolon-0.4.3)
Redis 4.0.11 (Helm chart redis-ha-3.0.0)

**Comment by shinji62** [comment]
@Morlackx I found my issue with the connexion.

Seems that `redis-ha` do not failover to other master if needed and Harbor do not use sentinel.
So I move to GCP memory store (because I am using GKE) and don't have any issue anymore.

I try to avoid database on k8s too, to I use cloudSQL for postgres. and of course GCS object storage.

**Comment by Morlackx** [comment]
@shinji62 Nice! I'll try with a redis single instance an I let you know.

**Comment by shinji62** [comment]
Ok let me go.

**Comment by Morlackx** [comment]
@shinji62 Definitely the issue was Redis. As you said, I will need a solution like Redis Sentinel or at least an haproxy with master check/health ontop of them. 

Thank you to point out this issue.

**Comment by shinji62** [comment]
@Morlackx Which is strange as redis-ha should normally provides that

**Comment by Morlackx** [comment]
@shinji62 I've tried again with redis-ha..and it is working as expected..I guess some wrong config by my side..probably.

**Comment by shinji62** [comment]
@Morlackx Will be fine until redis-ha failover ...

**Comment by ywk253100** [comment]
As explained in the document https://github.com/goharbor/harbor-helm/blob/master/docs/High%20Availability.md#configuration, the Redis client used by Harbor's upstream projects doesn't support Sentinel, Harbor can only work with a single entry point Redis.

**Comment by Morlackx** [comment]
@ywk253100 Thanks, I've missed that point. So, I've setup an haproxy that forward the harbor requests to the redis master as shown here [https://community.pivotal.io/s/article/How-to-setup-HAProxy-and-Redis-Sentinel-for-automatic-failover-between-Redis-Master-and-Slave-servers](url) .

The redis cluster address list is retrieved by haproxy querying the k8s DNS with the SRV record mapped by the stateful service ([https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#pod-identity](url)). But at this point haproxy gave me a segfault. I had to remove the "tcp-check connect" from the haproxy check.

Another issue I got is an EOF in harbor. For now, I've set all haproxy default timeouts to 0s and seems it is working now..but need to perform more tests before to claim it "stable".
